src/coreclr/jit/helperexpansion.cpp

```diff
@@ -478,9 +478,8 @@ bool Compiler::fgExpandThreadLocalAccessForCall(BasicBlock** pBlock, Statement*
         return false;
     }
 
-#if defined(FEATURE_READYTORUN)
-    assert(!"Unsupported scenario of optimizing TLS access for ReadyToRun");
-#endif
+    assert(!opts.IsReadyToRun());
+
 #if defined(TARGET_ALPINE_LINUX)
     assert(!"Unsupported scenario of optimizing TLS access on Linux Alpine");
 #endif
```

Comment:
We don't build any JIT with this defined so I don't think it makes sense to have this one on the JIT side either.

Reply:
That's true actually. will remove.